src/nifreeze/model/pet.py

```diff
@@ -101,26 +96,22 @@ def fit(self, data, **kwargs):
         AT = A.T
         ATdotA = AT @ A
 
-        # One single CPU - linear execution (full model)
-        if n_jobs == 1:
-            self._coeff = np.array([cg(ATdotA, AT @ v)[0] for v in data])
-            return
-
         # Parallelize process with joblib
-        with Parallel(n_jobs=n_jobs) as executor:
+        with Parallel(n_jobs=n_jobs or min(cpu_count() or 1, 8)) as executor:
             results = executor(delayed(cg)(ATdotA, AT @ v) for v in data)
 
         self._coeff = np.array([r[0] for r in results])
 
-    def predict(self, index=None, **kwargs):
+    def fit_predict(self, index: int | None = None, **kwargs):
```

Comment:
Preparing for #120 (this model is thought to be fit only once as it is designed)